source/grid/grid_tools.cc

```diff
@@ -2553,7 +2553,7 @@ namespace GridTools
     if (matrix.m() == spacedim)
       for (int i = 0; i < spacedim; ++i)
         for (int j = 0; j < spacedim; ++j)
-          distance(i) = matrix(i,j) * point1(j);
+          distance(i) += matrix(i,j) * point1(j);
```

Comment:
periodicity_02.cc uses this line, but there it doesn't matter whether the matrix is {{0,1},{-1,0}} or {{0,1},{0,0}}. Apparently, make_periodicity_constraints with a rotation matrix has not been used for cases much different to the test case.